### ISSUE TITLE
Issue rendering scene immediately after switching the root scene

### ISSUE THREAD

**Issue by axelson** [boydm/scenic#108]
### Checklist

- [x] Check other issues and make sure that it is not reported yet.

### Versions and Environment

Elixir:

```bash
# elixir -v
Erlang/OTP 21 [erts-10.0.7] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Elixir 1.7.3 (compiled with Erlang/OTP 21)
```

Erlang:

```bash
# erl -v
Erlang/OTP 21 [erts-10.0.7] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Eshell V10.0.7  (abort with ^G)
```

Scenic:

```text
# mix deps | grep scenic
* scenic_sensor 0.7.0 (Hex package) (mix)
  locked at 0.7.0 (scenic_sensor) a12aed75
* scenic 0.9.0 (Hex package) (mix)
  locked at 0.9.0 (scenic) 129594f9
* scenic_clock 0.8.0 (Hex package) (mix)
  locked at 0.8.0 (scenic_clock) fc144430
```

OS:

```text
Arch Linux
```

### Steps to reproduce

1. In one scene run `ViewPort.set_root/2`
2. In the newly run scene, render the graph with `push_graph` in the `init/1` callback

### Expected Behavior

The new scene/graph should be rendered

### Actual Behavior

I only see a black screen (unless I add a delay to the `push_graph` call of around 100ms)

### Additional Comments

I'm not sure if I'm misunderstanding something (although that is definitely possible). For testing I'm actually using the Splash Scene from `scenic_new` so I don't think it is an issue with the scene transition (although I did look into it, which is why I was looking for the existence of `release_graph`).

**Comment by fantypants** [comment]
have you made sure your VP is set correctly? I had the same issue but found I needed to pass the VP state into the set_root/2 function

```def filter_event({:click, :btn_balance}, _, %{viewport: vp} = state) do
    ViewPort.set_root(vp, {ElixWallet.Scene.Balance, nil})
    {:continue, {:click, :btn_balance}, state}
  end```

**Comment by boydm** [comment]
The first parameter needs to be the pid of the viewport you are setting the root on. It is passed to your scene as one of the options in the options list (2nd parameter of the init function).

```
def init(_, opts) do
  ViewPort.set_root(opts[:viewport], ... )
end
```

**Comment by axelson** [comment]
@boydm I am passing the viewport pid correct and my new scene's `init/2` is being called (which calls `push_graph`), however the graph does not appear to be rendered to the screen. Perhaps I should make a repo with a reproduction?

**Comment by boydm** [comment]
That would be most helpful.

Have you tried on any other OS's? Is hardish for me to repo on Arch Linux...

**Comment by axelson** [comment]
Yeah, I'll give it a shot on Mac as well.

**Comment by axelson** [comment]
Okay, here's the semi-reproducible example: https://github.com/axelson/scenic_repro_draw

I can only reproduce it on Arch Linux, and not on Nerves or OS X. If I set the fill or send `push_graph` after 100ms it works, but otherwise I just get a black screen.

**Comment by boydm** [comment]
This should be fixed in v0.11. @axelson you agree?

**Comment by axelson** [comment]
I expect that it is. If I experience it again I'll re-open (or most likely just open a new issue)
